airflow/api_connexion/endpoints/dag_run_endpoint.py

```diff
@@ -116,43 +115,12 @@ def get_upstream_dataset_events(
             "DAGRun not found",
             detail=f"DAGRun with DAG ID: '{dag_id}' and DagRun ID: '{dag_run_id}' not found",
         )
-    events = _get_upstream_dataset_events(dag_run=dag_run, session=session)
+    events = dag_run.dataset_events
```

Comment:
This seems to be a different value. The previous function retrieves events that triggered `dag_run`, but `dag_run.dataset_events` is the events emitted by the run (if I understand the logic correctly)

Reply:
The only events on a DagRun are the events that that dag run "consumes" -- DagRuns don't produce datasets.

Perhaps the names should change, but the tests still pass (though I did change them, but I didn't change what dag the event is generated against)

Reply:
Without prior knowledge the name does imply events *from* the DagRun, not *causing* the DagRun, at least to my eyes. `consumed_dataset_events` perhaps?

Reply:
`consumed_dataset_events` it is.